### PR TITLE
Bug 1129665 - Convert testOpenInBackground

### PR DESCRIPTION
**Pull request by galgeek**
WIP, not sure how to find/access context menu, and what about middle click? And should opening a tab by context menu or middle click get added to firefox_puppeteer?

### CODE REVIEW

File: firefox_ui_tests/functional/tabbed_browsing/test_open_in_background.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    def tearDown(self):
+        try:
+            self.browser.tabbar.close_all_tabs([self.browser.tabbar.tabs[0]])
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    def test_open_in_background(self):
+        tabbar = self.browser.tabbar
+        self.browser.navbar.locationbar.load_url(self.test_url)
```
Comment (whimboo) [inline firefox_ui_tests/functional/tabbed_browsing/test_open_in_background.py:28]: We need the page being loaded here so we should use `navigate()` as usual.

File: firefox_ui_tests/functional/tabbed_browsing/test_open_in_background.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            link = self.marionette.find_element(By.NAME, 'link_1')
+            link.context_click()
+            # js: var contextMenuItem = this.getElement({type: "openLinkInNewTab"}); ???
+            context_menu_item = self.marionette.find_element(By.ID, 'openLinkInNewTab')
+            context_menu_item.click()
+
+            self.assertEqual(len(tabbar.tabs), 2)
+            # if we opened sucessfully in background, we should still be on 1st tab
+            self.assertEqual(tabbar.tabs[0].handle, self.marionette.current_window_handle)
+            tabbar.switch_to(1)
```
Comment (whimboo) [inline firefox_ui_tests/functional/tabbed_browsing/test_open_in_background.py:44]: I would not use window handles here but the tabs. `selected_tab == tabs[0]`

File: firefox_ui_tests/functional/tabbed_browsing/test_open_in_background.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            self.assertEqual(tabbar.tabs[0].handle, self.marionette.current_window_handle)
+
+            link = self.marionette.find_element(By.NAME, 'link_2')
+            link.context_click()
+            # js: var contextMenuItem = this.getElement({type: "openLinkInNewTab"}); ???
+            context_menu_item = self.marionette.find_element(By.ID, 'openLinkInNewTab')
+            context_menu_item.click()
+        
+            link = self.marionette.find_element(By.NAME, 'link_3')
+            link.context_click()
```
Comment (whimboo) [inline firefox_ui_tests/functional/tabbed_browsing/test_open_in_background.py:56]: Make sure you also use a middle click.

File: firefox_ui_tests/functional/tabbed_browsing/test_open_in_background.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+            tabbar.switch_to(1)
+            Wait(self.marionette).until(expected.element_present(lambda m:
+                                        m.find_element(By.ID, '1')))
+            tabbar.switch_to(2)
+            Wait(self.marionette).until(expected.element_present(lambda m:
+                                        m.find_element(By.ID, '2')))
+            tabbar.switch_to(3)
+            Wait(self.marionette).until(expected.element_present(lambda m:
+                                        m.find_element(By.ID, '3')))
```
Comment (whimboo) [inline firefox_ui_tests/functional/tabbed_browsing/test_open_in_background.py:71]: Maybe a mapping as what we have in the Mozmill test would make things better understandable and would reduce code duplication.

File: firefox_ui_tests/functional/tabbed_browsing/test_open_in_background.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+            self.browser.tabbar.close_all_tabs([self.browser.tabbar.tabs[0]])
```
Comment (whimboo) [inline firefox_ui_tests/functional/tabbed_browsing/test_open_in_background.py:73]: It's already in tearDown, so not necessary here.

**Comment by whimboo** [comment]
Barbara, is something blocking us to continue on this PR?

**Comment by galgeek** [comment]
@whimboo we were waiting for some work on the menu library. I still see a menu-related PR open, but maybe what's needed here, for a context menu, is done?

**Comment by whimboo** [comment]
Oh right! I missed that part. So no, this is not done yet. I hope that @tareqak will find the time to update his PR about the menu class refactoring soon.

**Pull request closed without merging**